### ISSUE TITLE
[a11y]: Table row index number is not accurately conveyed to screen reader users

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#5676]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.44.0-rc.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Inconsistent table navigation for screen reader users when attempting to move between rows using keyboard controls. The screen reader announces an incorrect row index, leading to confusion and navigation difficulties.

**Recommended fix**
Screen reader expected to convey the actual /accurate index numbers of the rows. While navigating, ensure screen readers announce the correct row and column indexes.

**User impact**
If the table structure is not conveyed properly, it’s difficult for screen reader users to navigate through the table content and correlate data.

Wrong index announcement of table rows, can significantly hinder users who rely on screen readers to access and interact with the table's content.

### WCAG 2.1 Violation

1.3.1. Info & Relationships

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-datagrid--infinite-scroll&globals=viewport:basic

### Steps to reproduce

To reproduce this defect:

1) Use a screen reader and navigate to a table on the webpage.
2) Try moving up or down using arrow keys. The announced row number might not correspond to the actual row being focused on.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by matthewgallo** [comment]
Haven't been able to reproduce with voice over. I also tested the [example tables from w3](https://www.w3.org/WAI/tutorials/tables/one-header/) and the announce the first row as being the second row because it counts the row of table headers as the first row. I'll reference the recording Raghu did for Datagrid to see if there is any other difference.
